### ISSUE TITLE
Sparse GP tests fail nondeterministically

### ISSUE THREAD

**Issue by fritzo** [uber/pyro#756]
The following test fails when run alone:
```sh
pytest -v tests/contrib/gp/test_models.py::test_forward_sgpr   # FAILS
```
however it passes when run as part of the whole file
```sh
pytest -v tests/contrib/gp/test_models.py  # PASSES
```
As a result travis-ci tests pass but local `make test` fails (because it parallelizes)

**Comment by fritzo** [comment]
@fehiepsi could you take a look if you have time? I'm not sure why the test fails when run alone.

**Comment by fehiepsi** [comment]
@fritzo I fixed the bug at #758. Have no idea why it passes when run as a whole @@. For the bug: I have used `torch.ones(2)` instead of `torch.Tensor([2])` >"<.

**Comment by fritzo** [comment]
Closed by #758
